README.md

```diff
@@ -1,411 +1,492 @@
+        - [new Response([body[, options]])](#new-responsebody-options)
+        - [response.ok](#responseok)
+    - [Class: Headers](#class-headers)
+        - [new Headers([init])](#new-headersinit)
+    - [Interface: Body](#interface-body)
+        - [body.body](#bodybody)
+        - [body.bodyUsed](#bodybodyused)
+        - [body.arrayBuffer()](#bodyarraybuffer)
+        - [body.blob()](#bodyblob)
+        - [body.json()](#bodyjson)
+        - [body.text()](#bodytext)
+        - [body.buffer()](#bodybuffer)
+        - [body.textConverted()](#bodytextconverted)
+    - [Class: FetchError](#class-fetcherror)
+- [License](#license)
+- [Acknowledgement](#acknowledgement)
+
+<!-- /TOC -->
+
+## Motivation
+
+Instead of implementing `XMLHttpRequest` in Node.js to run browser-specific [Fetch polyfill](https://github.com/github/fetch), why not go from native `http` to `fetch` API directly? Hence `node-fetch`, minimal code for a `window.fetch` compatible API on Node.js runtime.
+
+See Matt Andrews' [isomorphic-fetch](https://github.com/matthew-andrews/isomorphic-fetch) or Leonardo Quixada's [cross-fetch](https://github.com/lquixada/cross-fetch) for isomorphic usage (exports `node-fetch` for server-side, `whatwg-fetch` for client-side).
+
+
+## Features
+
+- Stay consistent with `window.fetch` API.
+- Make conscious trade-off when following [whatwg fetch spec][whatwg-fetch] and [stream spec](https://streams.spec.whatwg.org/) implementation details, document known difference.
+- Use native promise, but allow substituting it with [insert your favorite promise library].
+- Use native stream for body, on both request and response.
+- Decode content encoding (gzip/deflate) properly, convert `res.text()` output to UTF-8 optionally.
+- Useful extensions such as timeout, redirect limit, response size limit, [explicit errors][ERROR-HANDLING.md] for troubleshooting.
+
+
+## Difference from client-side fetch
+
+- See [Known Differences][LIMITS.md] for details.
+- If you happen to use a missing feature that `window.fetch` offers, feel free to open an issue.
+- Pull requests are welcomed too!
+
+
+## Installation
+
+Current stable release (`2.x`)
+
+```sh
+$ npm install node-fetch --save
+```
+
+## Loading and configuring the module
+We suggest you load the module via `require`, pending the stabalizing of es modules in node:
+```javascript
+const fetch = require('node-fetch');
+
+```
+If you are using a Promise library other than native, set it through fetch.Promise:
+```javascript
+const Bluebird = require('bluebird');
+
+fetch.Promise = Bluebird;
+```
+
+## Common Usage
+
+NOTE: The documentation below is up-to-date with `2.x` releases, [see `1.x` readme](https://github.com/bitinn/node-fetch/blob/1.x/README.md), [changelog](https://github.com/bitinn/node-fetch/blob/1.x/CHANGELOG.md) and [2.x upgrade guide][UPGRADE-GUIDE.md] for the differences.
+
+#### Plain text or HTML
+```javascript
+fetch('https://github.com/')
+    .then(res => res.text())

```

Comment:
Style-wise, the code base uses tabs for indenting. I understand why you changed it to spaces, but I feel two spaces is more common in the JavaScript world anyway.